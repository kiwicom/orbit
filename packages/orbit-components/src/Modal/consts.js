// @flow
export const SIZES = {
  EXTRASMALL: "extraSmall",
  SMALL: "small",
  NORMAL: "normal",
  LARGE: "large",
  EXTRALARGE: "extraLarge",
};

export const CLOSE_BUTTON_DATA_TEST = "ModalCloseButton";
