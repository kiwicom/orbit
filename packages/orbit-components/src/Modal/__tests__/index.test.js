// @flow
import * as React from "react";
import { render, screen, act } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import Modal from "../index";
import { CLOSE_BUTTON_DATA_TEST } from "../consts";
import ModalHeader from "../ModalHeader";
import ModalSection from "../ModalSection";
import ModalFooter from "../ModalFooter";
import Illustration from "../../Illustration";
import useMediaQuery from "../../hooks/useMediaQuery";

jest.useFakeTimers("modern");

jest.mock("../../hooks/useMediaQuery", () => {
  return jest.fn(() => ({ isLargeMobile: true }));
});

const mockUseMediaQuery: JestMockFn<[], $Shape<$Call<typeof useMediaQuery>>> = useMediaQuery;

beforeEach(() => {
  jest.clearAllTimers();
});

describe("Modal", () => {
  it("should have expected DOM output", () => {
    render(
      <Modal dataTest="container">
        <ModalHeader
          dataTest="header"
          title="modal title"
          description="modal description"
          illustration={<Illustration name="Accommodation" dataTest="illustration" />}
        >
          header content
        </ModalHeader>
        <ModalSection dataTest="section">section content</ModalSection>
        <ModalFooter dataTest="footer">footer content</ModalFooter>
      </Modal>,
    );

    expect(screen.getByRole("dialog")).toBeInTheDocument();
    expect(screen.getByTestId("container")).toBeInTheDocument();
    expect(screen.getByTestId("header")).toBeInTheDocument();
    expect(screen.getByTestId("section")).toBeInTheDocument();
    expect(screen.getByTestId("footer")).toBeInTheDocument();
    expect(screen.getByTestId("illustration")).toBeInTheDocument();
    // title appears twice in the DOM, the second one is for the sticky header,
    // which is initially hidden and appears on scroll
    // when original title is out from the viewport
    expect(screen.getAllByText("modal title")).toHaveLength(2);
    expect(screen.getByText("modal description")).toBeInTheDocument();
    expect(screen.getByText("header content")).toBeInTheDocument();
    expect(screen.getByText("section content")).toBeInTheDocument();
    expect(screen.getByText("footer content")).toBeInTheDocument();
  });

  it("should be able to focus content", async () => {
    render(
      <Modal>
        <input />
      </Modal>,
    );

    act(() => {
      jest.runOnlyPendingTimers();
    });
    expect(screen.getByRole("dialog")).toHaveFocus();
    userEvent.tab(screen.getByRole("dialog"));
    act(() => {
      jest.runOnlyPendingTimers();
    });
    expect(screen.getByRole("textbox")).toHaveFocus();
  });

  it("should expose ref scrolling API", () => {
    const ref = React.createRef<React.ElementRef<typeof Modal>>();
    render(
      <Modal ref={ref} dataTest="test">
        content
      </Modal>,
    );

    ref.current?.setScrollPosition(20);
    const scrollPosition = ref.current?.getScrollPosition();
    expect(scrollPosition).toBe(20);
  });

  it("should switch scrolling container based on the breakpoint", () => {
    const modalRef = React.createRef<React.ElementRef<typeof Modal>>();
    const scrollingElement = React.createRef<HTMLElement>();
    function scrollingElementRef(node) {
      scrollingElement.current = node;
    }
    mockUseMediaQuery.mockImplementation(() => ({ isLargeMobile: true }));
    const { rerender } = render(
      <Modal ref={modalRef} scrollingElementRef={scrollingElementRef}>
        content
      </Modal>,
    );
    expect(scrollingElement.current).toBe(modalRef.current?.modalBody.current);
    mockUseMediaQuery.mockImplementation(() => ({ isLargeMobile: false }));
    rerender(
      <Modal ref={modalRef} scrollingElementRef={scrollingElementRef}>
        content
      </Modal>,
    );
    expect(scrollingElement.current).toBe(modalRef.current?.modalContent.current);
    // default mocked implementation
    mockUseMediaQuery.mockImplementation(() => ({ isLargeMobile: true }));
  });

  it("should call callback function when clicking on close button", () => {
    const onClose = jest.fn();
    render(
      <Modal hasCloseButton onClose={onClose}>
        content
      </Modal>,
    );

    userEvent.click(screen.getByTestId(CLOSE_BUTTON_DATA_TEST));
    expect(onClose).toBeCalled();
  });

  it("should have fixed header background overlay when needed", () => {
    const { rerender } = render(
      <Modal hasCloseButton onClose={() => {}}>
        content
      </Modal>,
    );
    screen.getByTestId("CloseContainer");
    rerender(
      <Modal>
        <ModalHeader title="title" />
      </Modal>,
    );
    screen.getByTestId("CloseContainer");
  });
});
