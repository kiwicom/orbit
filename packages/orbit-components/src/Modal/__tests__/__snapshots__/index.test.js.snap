// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly 1`] = `
.c18 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 16px;
  font-weight: 400;
  color: #252A31;
  line-height: 22px;
  text-align: left;
  margin: 0;
}

.c18 a:not(.c40),
.c18 a:not(.c40):link,
.c18 a:not(.c40):visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c18 a:not(.c40):hover,
.c18 a:not(.c40):active,
.c18 a:not(.c40):focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #009882;
}

.c6 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c33 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: currentColor;
  -webkit-transition: background 0.15s ease-in-out,box-shadow 0.15s ease-in-out,color 0.15s ease-in-out;
  transition: background 0.15s ease-in-out,box-shadow 0.15s ease-in-out,color 0.15s ease-in-out;
}

.c8 > svg {
  width: 24px;
  height: 24px;
}

.c34 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 8px 0 0;
  color: currentColor;
  -webkit-transition: background 0.15s ease-in-out,box-shadow 0.15s ease-in-out,color 0.15s ease-in-out;
  transition: background 0.15s ease-in-out,box-shadow 0.15s ease-in-out,color 0.15s ease-in-out;
}

.c34 > svg {
  width: 24px;
  height: 24px;
}

.c35 {
  display: inline-block;
  width: 100%;
  text-align: left;
}

.c38 {
  display: inline-block;
  width: 100%;
}

.c5 {
  height: 44px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  max-width: 100%;
  background: transparent;
  color: #252A31!important;
  border: 0;
  padding: 0;
  border-radius: 6px;
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 1.4;
  cursor: pointer;
  -webkit-transition: all 0.15s ease-in-out !important;
  transition: all 0.15s ease-in-out !important;
  outline: 0;
  width: 44px;
}

.c5 .c7 {
  color: #252A31;
}

.c5 .c41 {
  width: 24px;
  height: 24px;
}

.c5:hover {
  background: #E5EAEF;
  color: #181B20!important;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:hover .c7 {
  color: #181B20 !important;
}

.c5:active {
  background: #D6DEE6;
  color: #0B0C0F!important;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:active .c7 {
  color: #0B0C0F !important;
}

.c5:focus {
  background: #D6DEE6;
  color: #0B0C0F!important;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:focus .c7 {
  color: #0B0C0F !important;
}

.c5:focus:not(:focus-visible) {
  box-shadow: none;
  background: transparent;
  color: #0B0C0F!important;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:focus:not(:focus-visible) .c7 {
  color: #0B0C0F !important;
}

.c5:-moz-focusring,
.c5:focus-visible {
  background: #D6DEE6;
  color: #0B0C0F!important;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:-moz-focusring .c7,
.c5:focus-visible .c7 {
  color: #0B0C0F !important;
}

.c32 {
  height: 44px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  max-width: 100%;
  background: #E8EDF1;
  color: #252A31!important;
  border: 0;
  padding: 0 16px 0 12px;
  border-radius: 6px;
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 1.4;
  cursor: pointer;
  -webkit-transition: all 0.15s ease-in-out !important;
  transition: all 0.15s ease-in-out !important;
  outline: 0;
  width: auto;
}

.c32 .c7 {
  color: #252A31;
}

.c32 .c41 {
  width: 24px;
  height: 24px;
}

.c32:hover {
  background: #DCE3E9;
  color: #181B20!important;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c32:hover .c7 {
  color: #181B20 !important;
}

.c32:active {
  background: #CAD4DE;
  box-shadow: inset 0 0 6px 3px rgba(37,42,49,0.08);
  color: #0B0C0F!important;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c32:active .c7 {
  color: #0B0C0F !important;
}

.c32:focus {
  box-shadow: 0 0 0 3px rgba(95,115,140,0.3);
  color: !important;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c32:focus .c7 {
  color: #0B0C0F !important;
}

.c32:focus:not(:focus-visible) {
  box-shadow: none;
  background: #E8EDF1;
  color: !important;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c32:focus:not(:focus-visible) .c7 {
  color: #0B0C0F !important;
}

.c32:-moz-focusring,
.c32:focus-visible {
  box-shadow: 0 0 0 3px rgba(95,115,140,0.3);
  color: !important;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c32:-moz-focusring .c7,
.c32:focus-visible .c7 {
  color: #0B0C0F !important;
}

.c37 {
  height: 44px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  max-width: 100%;
  background: #00A991;
  color: #FFFFFF!important;
  border: 0;
  padding: 0 16px;
  border-radius: 6px;
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 1.4;
  cursor: pointer;
  -webkit-transition: all 0.15s ease-in-out !important;
  transition: all 0.15s ease-in-out !important;
  outline: 0;
  width: 100%;
}

.c37 .c7 {
  color: #FFFFFF;
}

.c37 .c41 {
  width: 24px;
  height: 24px;
}

.c37:hover {
  background: #009882;
  color: #FFFFFF!important;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c37:hover .c7 {
  color: #FFFFFF !important;
}

.c37:active {
  background: #008F7B;
  box-shadow: inset 0 0 6px 3px rgba(37,42,49,0.15);
  color: #FFFFFF!important;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c37:active .c7 {
  color: #FFFFFF !important;
}

.c37:focus {
  box-shadow: 0 0 0 3px rgba(0,169,145,0.5);
  color: !important;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c37:focus .c7 {
  color: #FFFFFF !important;
}

.c37:focus:not(:focus-visible) {
  box-shadow: none;
  background: #00A991;
  color: !important;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c37:focus:not(:focus-visible) .c7 {
  color: #FFFFFF !important;
}

.c37:-moz-focusring,
.c37:focus-visible {
  box-shadow: 0 0 0 3px rgba(0,169,145,0.5);
  color: !important;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c37:-moz-focusring .c7,
.c37:focus-visible .c7 {
  color: #FFFFFF !important;
}

.c31 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  box-sizing: border-box;
  padding: 0 8px 0 0;
}

.c36 {
  -webkit-flex: 1 1 100%;
  -ms-flex: 1 1 100%;
  flex: 1 1 100%;
  box-sizing: border-box;
  padding: 0 8px 0 0;
}

.c29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 800;
  width: 100%;
  background-color: #FFFFFF;
  padding: 0 16px 16px;
  box-sizing: border-box;
  -webkit-transition: box-shadow 0.15s ease-in-out;
  transition: box-shadow 0.15s ease-in-out;
}

.c29 .c30:last-of-type {
  padding: 0;
}

.c27 {
  width: 100%;
  padding: 24px 16px;
  background-color: #F5F7F9;
  border-bottom: 1px solid #EFF2F5;
  box-sizing: border-box;
}

.c27:first-of-type {
  border-top: 1px solid #EFF2F5;
  border-top-left-radius: 12px;
  border-top-right-radius: 12px;
  margin-top: 24px;
}

.c27:last-of-type {
  border-bottom: 1px solid #EFF2F5;
  border-bottom-left-radius: 12px;
  border-bottom-right-radius: 12px;
}

.c27:last-of-type ~ .c28 {
  margin-top: 16px;
}

.c27:last-of-type:not(:last-child) {
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.c15 {
  font-size: 22px;
  font-weight: 500;
  line-height: 28px;
  color: #252A31;
}

.c13 {
  margin-top: 8px;
}

.c13 .c14 {
  padding-right: 32px;
}

.c16 {
  margin-top: 8px;
}

.c11 {
  width: 100%;
  display: block;
  padding: 40px 16px;
  border-top-left-radius: 12px;
  border-top-right-radius: 12px;
  box-sizing: border-box;
  background-color: #F5F7F9;
}

.c11 ~ .c26:first-of-type {
  border-top: 1px solid #EFF2F5;
  border-top-left-radius: 0;
  border-top-right-radius: 0;
  margin-top: 0!important;
}

.c25 {
  display: inline-block;
  position: fixed;
  visibility: hidden;
  height: 52px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  top: 16px;
  right: 48px;
  left: 0;
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-weight: 700;
  font-size: 18px;
  color: #252A31;
  line-height: 52px;
  box-sizing: border-box;
  padding: 0 0 0 24px;
  opacity: 0;
  -webkit-transition: top 0.15s ease-in-out,opacity 0.15s ease-in-out,visibility 0.15s ease-in-out;
  transition: top 0.15s ease-in-out,opacity 0.15s ease-in-out,visibility 0.15s ease-in-out;
  z-index: 800;
}

.c19 {
  margin-top: 32px;
}

.c9 {
  width: 24px;
  height: 24px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  fill: currentColor;
}

.c0 {
  width: 100%;
  height: 100%;
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 800;
  box-sizing: border-box;
  outline: none;
  overflow-x: hidden;
  background-color: rgba(0,0,0,0.5);
  -webkit-overflow-scrolling: auto;
}

.c1 {
  box-sizing: border-box;
  min-height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin: 0 auto;
  position: fixed;
  width: 100%;
  border-top-left-radius: 12px;
  border-top-right-radius: 12px;
  -webkit-transition: top 0.3s ease-in-out;
  transition: top 0.3s ease-in-out;
  top: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  position: absolute;
  top: 0;
  right: 0;
  z-index: 800;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  height: 52px;
  width: 100%;
  max-width: 1280px;
  border-top-left-radius: 12px;
  border-top-right-radius: 12px;
  -webkit-transition: box-shadow 0.15s ease-in-out,background-color 0.15s ease-in-out;
  transition: box-shadow 0.15s ease-in-out,background-color 0.15s ease-in-out;
  pointer-events: none;
}

.c3 + .c26:first-of-type {
  padding-top: 52px;
  border-top: 0;
  margin: 0;
}

.c3 .c4 {
  pointer-events: auto;
  margin-right: 4px;
}

.c3 .c4 svg {
  -webkit-transition: color 0.15s ease-in-out;
  transition: color 0.15s ease-in-out;
  color: #5F738C;
}

.c3 .c4:hover svg {
  color: #52647A;
}

.c3 .c4:active svg {
  color: #465567;
}

.c39 .c10 {
  margin-bottom: 32px;
}

.c2 {
  position: absolute;
  box-sizing: border-box;
  border-top-left-radius: 12px;
  border-top-right-radius: 12px;
  background-color: #FFFFFF;
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  width: 100%;
  max-height: calc( 100% - 32px - 0px );
  bottom: 32px;
  box-shadow: 0px 12px 24px -4px rgba(37,42,49,0.24),0px 8px 60px 0px rgba(37,42,49,0.32);
  overflow-y: auto;
  overflow-x: hidden;
}

.c12 {
  display: inline-block;
  margin: auto 0;
  max-height: 90px;
  max-width: 100%;
  background-color: transparent;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c21 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-bottom: 4px;
}

.c21:last-child,
.c21:last-of-type {
  margin: 0;
}

.c21 .c17 {
  line-height: inherit;
  font-size: inherit;
}

.c22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 8px 0 0;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 20px;
}

.c22 .c42 {
  height: 16px;
  width: 16px;
}

.c22 .c42 img {
  height: 16px;
  width: 16px;
}

.c22 svg {
  height: 16px;
  width: 16px;
}

.c23 {
  width: 100%;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 14px;
  line-height: 20px;
  color: #252A31;
  list-style: none;
  padding: 0;
  margin: 0;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c6 {
    min-width: 100%;
    max-width: 1px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c33 {
    min-width: 100%;
    max-width: 1px;
  }
}

@media (min-width:768px) {
  .c5 {
    border-radius: 3px;
  }
}

@media (min-width:768px) {
  .c32 {
    border-radius: 3px;
  }
}

@media (min-width:768px) {
  .c37 {
    border-radius: 3px;
  }
}

@media (min-width:576px) {
  .c31 {
    -webkit-flex: none;
    -ms-flex: none;
    flex: none;
  }
}

@media (min-width:576px) {
  .c36 {
    -webkit-flex: none;
    -ms-flex: none;
    flex: none;
  }
}

@media (max-width:575px) {
  .c29 .c4 {
    font-size: 14px;
    height: 44px;
  }
}

@media (min-width:576px) {
  .c29 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    border-bottom-left-radius: 9px;
    border-bottom-right-radius: 9px;
  }
}

@media (min-width:576px) {
  .c27 {
    padding: 40px;
  }

  .c27:first-of-type {
    margin-top: 40px;
    border-top-left-radius: 9px;
    border-top-right-radius: 9px;
  }

  .c27:last-of-type {
    border-bottom-left-radius: 9px;
    border-bottom-right-radius: 9px;
  }

  .c27:last-of-type ~ .c28 {
    margin-top: 0;
  }
}

@media (min-width:576px) {
  .c15 {
    font-size: 28px;
    font-weight: 700;
    line-height: 36px;
  }
}

@media (min-width:992px) {
  .c13 .c14 {
    padding: 0;
  }
}

@media (min-width:576px) {
  .c11 {
    padding: 40px;
  }

  .c11 ~ .c26:first-of-type {
    border-top-left-radius: 0;
    border-top-right-radius: 0;
  }
}

@media (min-width:576px) {
  .c25 {
    left: auto;
    right: auto;
    padding: 0;
  }
}

@media (min-width:576px) {
  .c0 {
    overflow-y: auto;
    padding: 40px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c0 {
    position: -ms-page;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c1 {
    height: 1px;
  }
}

@media (min-width:576px) {
  .c1 {
    position: relative;
    top: 0;
    max-width: 1280px;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (min-width:576px) {
  .c3 {
    border-radius: 0;
  }
}

@media (min-width:576px) {
  .c39 .c26:last-of-type {
    padding-bottom: 40px;
    margin-bottom: 0px;
  }

  .c39 .c26:last-of-type::after {
    content: none;
  }

  .c39 .c10 {
    margin-bottom: 0px;
  }

  .c39 .c28 {
    padding: 32px 40px!important;
    max-width: 1280px;
  }

  .c39 .c24 {
    top: -40px;
    width: calc(0px - 48px - 40px);
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c39 .c28 {
    position: -ms-page;
  }
}

@media (min-width:576px) and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c39 .c28 {
    position: fixed;
  }
}

@media (min-width:576px) {
  .c2 {
    position: relative;
    bottom: auto;
    border-radius: 9px;
    padding-bottom: 0;
    height: auto;
    overflow: visible;
    max-height: 100%;
  }

  .c2 .c26:last-of-type {
    padding-bottom: 40px;
    margin-bottom: 0px;
  }

  .c2 .c26:last-of-type::after {
    content: none;
  }

  .c2 .c10 {
    margin-bottom: 0;
  }

  .c2 .c28 {
    padding: 32px 40px!important;
    max-width: 1280px;
    position: absolute;
    box-shadow: none;
  }

  .c2 .c24 {
    top: -40px;
    width: calc(0px - 48px - 40px);
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c2 .c28 {
    position: -ms-page;
  }
}

@media (min-width:576px) and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c2 .c28 {
    position: static;
  }

  .c2 .c26:last-of-type {
    margin-bottom: 0;
  }
}

<div>
  <div
    aria-labelledby="modalID50000-id-50000"
    aria-modal="true"
    class="c0"
    data-test="modal"
    role="dialog"
    tabindex="0"
  >
    <div
      class="c1"
      id="modalID50000-id-50000"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <button
            aria-label="Close"
            class="c4 c5"
            data-test="ModalCloseButton"
            type="button"
          >
            <div
              class="c6"
            >
              <div
                class="c7 c8"
              >
                <svg
                  class="c9"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17.655 6.333a.9.9 0 01.001 1.273l-4.103 4.108a.4.4 0 000 .566l4.103 4.109a.9.9 0 01.08 1.18l-.08.092a.9.9 0 01-1.273 0l-4.1-4.108a.4.4 0 00-.567 0l-4.1 4.107a.9.9 0 11-1.273-1.272l4.103-4.108a.4.4 0 000-.566L6.343 7.606a.9.9 0 01-.08-1.18l.081-.093a.9.9 0 011.273.001l4.099 4.106a.4.4 0 00.566 0l4.1-4.106a.9.9 0 011.273 0z"
                  />
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="c10 c11"
          data-test="header"
        >
          <img
            alt="Accommodation"
            class="c12"
            data-test="illustration"
            src="//images.kiwi.com/illustrations/0x90/Accommodation-Q85.png"
            srcset="//images.kiwi.com/illustrations/0x180/Accommodation-Q85.png 2x, //images.kiwi.com/illustrations/0x270/Accommodation-Q85.png 3x"
          />
          <div
            class="c13"
          >
            <div
              class="c14 c15"
            >
              Title of Modal
            </div>
            <div
              class="c16"
            >
              <div
                class="c17 c18"
              >
                Description of Modal
              </div>
            </div>
          </div>
          <div
            class="c19"
          >
            <ul
              class="c20"
            >
              <li
                class="c21"
              >
                <div
                  class="c22"
                  type="primary"
                >
                  <svg
                    class="c9"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17.76 16.954v.09h1.71c.09 0 .18.09.18.18v1.888c0 .989-.81 1.888-1.89 1.888H4.89C3.81 21 3 20.1 3 19.112v-9.98c0-1.08.81-1.889 1.89-1.889h12.87c1.08 0 1.89.81 1.89 1.888v1.888c0 .09-.09.18-.18.18h-1.71c-1.62 0-2.88 1.26-2.88 2.878s1.35 2.877 2.88 2.877zm2.25-4.586c.63.09.99.63.99 1.08v1.168c0 .63-.45 1.079-.99 1.169h-2.25c-.99 0-1.71-.81-1.71-1.708 0-.9.81-1.709 1.71-1.709h2.25zm-2.07-8.632l.45 2.068v.09c0 .09-.09.18-.18.18H7.05c-.09 0-.18-.09-.18-.18s.09-.18.18-.18l9.81-2.697c.45-.09.99.18 1.08.72z"
                    />
                  </svg>
                </div>
                <span
                  class="c23"
                >
                  To save you time, we have calculated your total possible refundable amount.
                </span>
              </li>
            </ul>
          </div>
          <div
            class="c24 c25"
          >
            Title of Modal
          </div>
        </div>
        <section
          class="c26 c27"
          data-test="section"
        >
          This is the content of the section
        </section>
        <div
          class="c28 c29"
          data-test="footer"
        >
          <div
            class="c30 c31"
          >
            <button
              class="c4 c32"
              type="button"
            >
              <div
                class="c33"
              >
                <div
                  class="c7 c34"
                >
                  <svg
                    class="c9"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.254 16.089c.369.333.398.904.065 1.274a.898.898 0 01-1.27.066l-5.303-4.795a.904.904 0 01.002-1.343l5.256-4.72a.898.898 0 011.271.07c.332.372.3.942-.07 1.275l-4.18 3.753a.4.4 0 000 .594l4.229 3.826z"
                    />
                  </svg>
                </div>
                <div
                  class="c35"
                >
                  Back
                </div>
              </div>
            </button>
          </div>
          <div
            class="c30 c36"
          >
            <button
              class="c4 c37"
              type="button"
            >
              <div
                class="c6"
              >
                <div
                  class="c38"
                >
                  Continue to Payment
                </div>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
